azure-client-runtime/src/main/java/com/microsoft/azure/v2/util/ServiceFutureUtil.java

```diff
@@ -0,0 +1,21 @@
+package com.microsoft.azure.v2.util;
+
+import com.microsoft.azure.v2.OperationStatus;
+import com.microsoft.rest.ServiceCallback;
+import com.microsoft.rest.ServiceFuture;
+import rx.Observable;
+import rx.Single;
+import rx.functions.Func1;
+
+public class ServiceFutureUtil {
+    public static <T> ServiceFuture<T> fromLRO(Observable<OperationStatus<T>> observable, ServiceCallback<T> callback) {
+        Single<T> single = observable.last().toSingle().map(new Func1<OperationStatus<T>, T>() {
+            @Override
+            public T call(OperationStatus<T> tOperationStatus) {
+                return tOperationStatus.result();
```

Comment:
This assumes that the operation completed successfully. Is that okay or is this only temporary?

Reply:
That's a good question. I think it depends on OperationStatus.result throwing if an error occurred, which might not be what we want.

Reply:
if `tOperationStatus.result()` throws exception the map operator will catch it and invoke `onError` in the downstream, which get translated to a call to `serviceFuture.setException(throwable)`.

If there is an error occurred during polling then `Observable<OperationStatus<T>>` should call `orError` on the downstream, which in this case again will get propagated and translated to a call to `serviceFuture.setException(throwable)`.

If above assumptions are correct then this implementation is correct, right?

Reply:
Yes, the implementation is correct.

Reply:
We determined that it's acceptable to assume that either the last OperationStatus should contain a non-null `.result()` or that something should have thrown somewhere earlier on causing Single.error to be returned to whoever finally uses this Single.

Basically, this code is OK, except the parameter name in the Func1 is wacky.